building-and-packaging/material/packaging_debian_and_spack_exercise.md

```diff
@@ -0,0 +1,99 @@
+# Packages with CPack and Spack
+
+This exercise is about packaging code with CPack and Spack. In this exercise we use adapted versions of the code from the previous week's exercise and create packages for it. The exercise is split into two parts: 1. CPack and 2. Spack. There are different GitHub repositories for the two subtasks.
+
+At the end of the exercise you find a section with hints and remarks. Make sure to check this section.
+
+Deadline: **Thursday, December 7th, 2021, 9:00**
+
+## 1. CPack
+
+**Attention/TODO**: Use [test repository](https://github.com/Simulation-Software-Engineering/cpack-exercise-test) for testing the exercise.
+
+The ultimate goals is to have a CMake/CPack configuration that allows us to generate packages for different targets. In this exercise we want to create a `tar.gz` package and a Debian `deb` package. We want to package this [HelloWorld example from GitHub](https://github.com/Simulation-Software-Engineering/cpack-exercise).
+
+When setting up the package make sure that the executable `cpackexample` and also the library `cpackexamplelib` are build and installed in appropriate locations. You should also ensure that the header files for the library are installed properly. Please package `cpackexamplelib` as **shared** library. The package should contain enough information (at least maintainer, contact address, project description, vendor, version number, homepage). If you do not want to use your actual name/email address, please use your GitHub username and make up an email address).
+
+For the Debian package, make sure that
+
+- the executable is stripped.
```

Comment:
I don't understand what you mean by "stripped" here.

Reply:
That is an executable for which [one has removed the debug symbols](https://unix.stackexchange.com/questions/2969/what-are-stripped-and-not-stripped-executables-in-unix). There is a CPack option for this that should be mentioned in the lecture. If not, I am adding it.

Reply:
I rephrased this part to make it more clear what is meant.